lib/galaxy/schema/notifications.py

```diff
@@ -0,0 +1,256 @@
+import json
+from datetime import datetime
+from enum import Enum
+from typing import (
+    Any,
+    List,
+    Optional,
+    Union,
+)
+
+from pydantic import (
+    AnyUrl,
+    Field,
+)
+from pydantic.utils import GetterDict
+from typing_extensions import (
+    Annotated,
+    Literal,
+)
+
+from galaxy.schema.fields import (
+    DecodedDatabaseIdField,
+    EncodedDatabaseIdField,
+)
+from galaxy.schema.schema import (
+    CreateTimeField,
+    Model,
+    UpdateTimeField,
+)
+
+
+class NotificationVariant(str, Enum):
+    """The notification variant communicates the intent or relevance of the notification."""
+
+    info = "info"
+    warning = "warning"
+    urgent = "urgent"
+
+
+# Register a new category by adding it to the corresponding Enum:
+# - NotificationCategory: these notification categories will be always received by the user.
+# - OptionalNotificationCategory: these notifications can be opt-out by the user.
+# Then register the content model further down below.
+
+
+class MandatoryNotificationCategory(str, Enum):
+    """These notification categories cannot be opt-out by the user.
+
+    The user will always receive notifications from these categories.
+    """
+
+    broadcast = "broadcast"
+
+
+class PersonalNotificationCategory(str, Enum):
+    """These notification categories can be opt-out by the user and will be
+    displayed in the notification preferences.
+    """
+
+    message = "message"
+    near_quota_limit = "near_quota_limit"
+    new_history_shared = "new_history_shared"
+    new_workflow_shared = "new_workflow_shared"
+    new_page_shared = "new_page_shared"
+    new_visualization_shared = "new_page_shared"
+    workflow_execution_completed = "workflow_execution_completed"
```

Comment:
```suggestion
    workflow_execution_completed = "workflow_execution_completed"
```
😍 I'd love to build the backend for that, just be aware we don't have a way to hook into that yet

Reply:
Ahh good to know, those categories are just potential ideas for the moment, I'll disable that one for now.

Reply:
near_quota_limit should also be disabled for now I assume (push back if I'm wrong). We present a quota bar to the user in the UI - the client could just look at this and decide what to do - rather than having to check and fire this every time we write any object in Galaxy. 

This is overall wonderful. I'd love to see the notifications implemented - the sharing stuff seems like a good start. There is also a job completion -> email hook in Galaxy that you can probably piggy back off of to build another useful thing pretty quickly.

Reply:
Thank you! That makes a huge lot of sense @jmchilton!

I'll look into the job completion hook :+1: